src/seccomp/mod.rs

```diff
@@ -0,0 +1,416 @@
+        Arch::ScmpArchMipsel64n32 => SCMP_ARCH_MIPSEL64N32,
+        Arch::ScmpArchPpc => SCMP_ARCH_PPC,
+        Arch::ScmpArchPpc64 => SCMP_ARCH_PPC64,
+        Arch::ScmpArchPpc64le => SCMP_ARCH_PPC64LE,
+        Arch::ScmpArchS390 => SCMP_ARCH_S390,
+        Arch::ScmpArchS390x => SCMP_ARCH_S390X,
+    }
+}
+
+pub fn initialize_seccomp(seccomp: &LinuxSeccomp) -> Result<()> {
+    if seccomp.flags.is_some() {
+        // runc did not support this, so let's skip it for now.
+        bail!("seccomp flags are not yet supported");
+    }
+
+    // TODO: fix default action error number. The spec repo doesn't have it yet.
+    let default_action = translate_action(&seccomp.default_action, None);
+    let mut ctx = FilterContext::default(default_action)?;
+
+    if let Some(architectures) = seccomp.architectures.as_ref() {
+        for arch in architectures {
+            let arch_token = translate_arch(arch);
+            ctx.add_arch(arch_token as u32)
+                .context("Failed to add arch to seccomp")?;
+        }
+    }
+
+    // The SCMP_FLTATR_CTL_NNP controls if the seccomp load function will set
+    // the new privilege bit automatically in prctl. Normally this is a good
+    // thing, but for us we need better control. Based on the spec, if OCI
+    // runtime spec doesn't set the no new privileges in Process, we should not
+    // set it here.  If the seccomp load operation fails without enough
+    // privilege, so be it. To prevent this automatic behavior, we unset the
+    // value here.
+    let ret = unsafe { seccomp_attr_set(ctx.ctx, scmp_filter_attr::SCMP_FLTATR_CTL_NNP, 0) };
+    if ret != 0 {
+        bail!(
+            "Failed to unset the no new privileges bit for seccomp: {}",
+            ret
+        );
+    }
+
+    if let Some(syscalls) = seccomp.syscalls.as_ref() {
+        for syscall in syscalls {
+            let action = translate_action(&syscall.action, syscall.errno_ret);
+            if action == default_action {
+                // When the action is the same as the default action, the rule is redundent. We can
+                // skip this here to avoid failing when we add the rules.
+                log::warn!(
+                    "Detect a seccomp action that is the same as the default action: {:?}",
+                    syscall
+                );
+                continue;
+            }
+
+            for name in &syscall.names {
+                let ret = translate_syscall(name);
+                if ret.is_err() {
+                    // If we failed to resolve the syscall by name, likely the kernel
+                    // doeesn't support this syscall. So it is safe to skip...
+                    log::warn!(
+                        "Failed to resolve syscall, likely kernel doesn't support this. {:?}",
+                        name
+                    );
+                    continue;
+                }
+
+                let syscall_number = translate_syscall(name)?;
```

Comment:
Do you have to call translate_syscall here again? Couldn't you do just
```suggestion
                let syscall_number = ret.unwrap()
```
given that you have have already checked for an error above?